src/server/auth/server/roles.go

```diff
@@ -174,26 +174,27 @@ func init() {
 	// Project related roles
 	projectViewer := registerRole(&auth.Role{
 		Name:          auth.ProjectViewer,
-		ResourceTypes: []auth.ResourceType{auth.ResourceType_PROJECT, auth.ResourceType_CLUSTER},
+		ResourceTypes: []auth.ResourceType{auth.ResourceType_CLUSTER, auth.ResourceType_PROJECT},
 		Permissions: []auth.Permission{
 			auth.Permission_PROJECT_LIST_REPO,
 		},
 	})
 
 	projectWriter := registerRole(&auth.Role{
 		Name:          auth.ProjectWriter,
-		ResourceTypes: []auth.ResourceType{auth.ResourceType_PROJECT, auth.ResourceType_CLUSTER},
+		ResourceTypes: []auth.ResourceType{auth.ResourceType_CLUSTER, auth.ResourceType_PROJECT},
 		Permissions: combinePermissions(projectViewer.Permissions, []auth.Permission{
-			auth.Permission_PROJECT_LIST_REPO,
+			auth.Permission_PROJECT_CREATE_REPO,
 		}),
 	})
 
 	projectOwner := registerRole(&auth.Role{
 		Name:          auth.ProjectOwner,
-		ResourceTypes: []auth.ResourceType{auth.ResourceType_PROJECT, auth.ResourceType_CLUSTER},
+		ResourceTypes: []auth.ResourceType{auth.ResourceType_CLUSTER, auth.ResourceType_PROJECT},
```

Comment:
should project owners be able to modify repo role bindings within their project?

Could you add a test that iterates over the main project, repo and cluster admin roles and tries to modify the role bindings for different resources to verify the behavior?

Reply:
Currently the ProjectOwner role can't modify repo role bindings because it doesn't have the `REPO_MODIFY_BINDINGS` permission. Should we make ProjectOwner inherit RepoOwner?

And yeap I'll add more tests!